File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Rules/IfElseWithRules/CS/IfElseWithRules/IfElseWorkflow.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System;
using System.Workflow.Activities;

namespace Microsoft.Samples.Workflow.IfElseActivityWithRules
{
    public sealed partial class IfElseWorkflow : SequentialWorkflowActivity
    {
        // This field is populated by the incoming workflow parameter and is used by the condition
        private int orderValue;

        public IfElseWorkflow()
        {
            InitializeComponent();
        }

        private void ManagerApprovalHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Get Manager Approval");
        }

        private void VPApprovalHandler(object sender, EventArgs e)
        {
            Console.WriteLine("Get VP Approval");
        }

        public int OrderValue
        {
            get
            {
                return this.orderValue;
            }
            set
            {
                this.orderValue = value;
            }
        }
    }
}

File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Rules/IfElseWithRules/VB/IfElseWithRules/IfElseWorkflow.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------
Imports System
Imports System.ComponentModel
Imports System.CodeDom
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

Namespace Microsoft.Samples.Rules.IfElseWithRules
    Public Class IfElseWorkflow
        Inherits SequentialWorkflowActivity

        ' This field is populated by the incoming workflow parameter and is used by the condition
        Private orderValueValue As Integer

        Public Property OrderValue() As Integer
            Get
                Return Me.orderValueValue
            End Get
            Set(ByVal value As Integer)
                Me.orderValueValue = value
            End Set
        End Property

        Private Sub ManagerApprovalHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Get Manager Approval")
        End Sub

        Private Sub VPApprovalHandler(ByVal sender As System.Object, ByVal e As System.EventArgs)
            Console.WriteLine("Get VP Approval")
        End Sub
    End Class
End Namespace
